CHANGES.md

```diff
@@ -103,10 +103,13 @@
   it was ignored if provided.
 * `TopCombineFn` disallow `compare` as its argument (Python) ([BEAM-7372](https://issues.apache.org/jira/browse/BEAM-7372)).
 * Drop support for Flink 1.10 ([BEAM-12281](https://issues.apache.org/jira/browse/BEAM-12281)).
+* Custom implementations of `TriggerFn` now require that `may_lose_data` be implemented ([BEAM-9487](https://issues.apache.org/jira/browse/BEAM-9487)).
```

Comment:
Same backward compatibility question applies. Would this break any existing TriggerFn's out there. 

And this is also a bit more severe then just adding a flag to workaround, because it will require code changes. Often times that is very hard and sometimes not possible. (E.g. An existing template where users do not have the sources easily accessible.) This type of change might be better waiting for Beam 3.0 (breaking changes) or happen very slowly.

Reply:
Ok, I removed `@abstractmethod` and gave a backwards-compatible default implementation. The note in CHANGES.md has been moved to the "New Features / Improvements" section, just so people know that the option exists.